client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -28,32 +28,42 @@ const DomainRegistrationSuggestion = React.createClass( {
 
 	render() {
 		const { suggestion, translate } = this.props,
-			isAdded = hasDomainInCart( this.props.cart, suggestion.domain_name ),
+			domain = suggestion.domain_name,
+			isAdded = hasDomainInCart( this.props.cart, domain ),
 			domainFlags = [];
+
 		let buttonClasses, buttonContent;
 
-		if ( suggestion.domain_name ) {
+		if ( domain ) {
 			const newTLDs = [];
+			const a8cTLDs = [ '.ca', '.de', '.fr' ];
+			const tld = domain.substring( domain.indexOf( '.' ) );
```

Comment:
No, this approach will actually support those better - since we only suggest root domains (not subdomains), everything after the _first_ dot should be a TLD. So, with the previous approach, we'd extract `uk` from `example.co.uk`, but with this one it will be `co.uk` 👍